pkg/controller/configmap/configmap_controller.go

```diff
@@ -96,28 +101,52 @@ func (r *ReconcileConfigMap) Reconcile(request reconcile.Request) (reconcile.Res
 		return reconcile.Result{}, err
 	}
 
-	// delete daemonSet pods when aide-conf configMap changes
-	podList := &corev1.PodList{}
-	err = r.client.List(context.TODO(), podList, client.InNamespace(instance.Namespace))
+	// only continue if the configmap received an update through the user-provided config
+	if _, ok := instance.Annotations["fileintegrity.openshift.io/updated"]; !ok {
+		return reconcile.Result{}, nil
+	}
+
+	ds := &appsv1.DaemonSet{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: common.WorkerDaemonSetName, Namespace: common.FileIntegrityNamespace}, ds)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			reqLogger.Info("no daemonSet pods found")
-			return reconcile.Result{}, nil
-		}
-		reqLogger.Error(err, "error getting pod list")
+		reqLogger.Error(err, "error getting worker daemonSet")
 		return reconcile.Result{}, err
 	}
 
-	for _, pod := range podList.Items {
-		if strings.HasPrefix(pod.Name, "aiderunner") {
-			reqLogger.Info("deleting pod", "Pod.Name", pod.Name)
-			podCopy := pod.DeepCopy()
-			delErr := r.client.Delete(context.TODO(), podCopy)
-			if delErr != nil {
-				return reconcile.Result{}, delErr
-			}
-		}
+	if err := triggerDaemonSetRollout(r.client, ds); err != nil {
```

Comment:
`triggerDaemonSetRollout` updates the DS. Is it correct to operate on the instance you got and not on its DeepCopy?